restapi/src/main/java/io/stargate/web/docsapi/service/json/DeadLeafCollector.java

```diff
@@ -0,0 +1,17 @@
+package io.stargate.web.docsapi.service.json;
+
+import java.util.Map;
+import java.util.Set;
+
+public interface DeadLeafCollector {
+
+  void addLeaf(String path, DeadLeaf leaf);
+
+  void addArray(String path);
+
+  void addAll(String path);
+
+  Map<String, Set<DeadLeaf>> getLeaves();
```

Comment:
Might be preferable to keep this at the impl. level to avoid requiring a particular data representation from all possible collectors.